File: SourceCode/api/helpers/email_validation.py
"""
Team Name: Red Team
Members:
 - Brayan Covarrubias
 - Matthew Rozendaal
 - Rashai Robertson
 - Tiffany Davidson
Description:
Reusable email format validation helper for the Moffat Bay Lodge API.
"""

import re

from api.models.validation_result import ValidationResult

# Matches "text@domain.tld" - requires a "." between the domain and a letters-only extension
EMAIL_PATTERN = re.compile(r"^[^\s@]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$")


def validate_email(email: str) -> ValidationResult:
    """Validate an email address's format and return a ValidationResult."""
    if email is None or not email.strip():
        return ValidationResult(is_valid=False, reason="Email address is required.")

    if EMAIL_PATTERN.match(email.strip()):
        return ValidationResult(is_valid=True, reason="Email address is valid.")

    return ValidationResult(
        is_valid=False,
        reason="Email address must be in the format name@domain.tld (e.g. you@example.com).",
    )

File: src/assets/js/emailValidation.js
/*
Team Name: Red Team
Members:
 - Brayan Covarrubias
 - Matthew Rozendaal
 - Rashai Robertson
 - Tiffany Davidson
Description:
Reusable input validation helpers. Each validator is a standalone function so
additional checks (beyond email) can be added here without touching existing ones.
*/

// Matches "text@domain.tld" - requires a "." between the domain and a letters-only extension
const EMAIL_PATTERN = /^[^\s@]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;

function isValidEmail(email) {
    return EMAIL_PATTERN.test(email.trim());
}

// Validates an email input; when showMessage is true, adds/removes a single validation message beneath it
function checkEmailInput(inputElement, showMessage) {
    const valid = isValidEmail(inputElement.value);
    const messageId = inputElement.id + "-validation-message";
    let messageElement = document.getElementById(messageId);

    if (valid) {
        if (messageElement) {
            messageElement.remove();
        }
        return true;
    }

    if (showMessage) {
        if (!messageElement) {
            messageElement = document.createElement("p");
            messageElement.id = messageId;
            messageElement.className = "input-validation-message";
            inputElement.insertAdjacentElement("afterend", messageElement);
        }
        messageElement.textContent = "Please enter a valid email address.";
    }

    return false;
}
